src/MUSI6106Exec/run_filtering.h

```diff
@@ -0,0 +1,97 @@
+//
+// Created by Vedant Kalbag on 13/02/22.
+//
+
+#ifndef MUSI6106_RUN_FILTERING_H
+#define MUSI6106_RUN_FILTERING_H
+using std::cout;
+using std::endl;
+Error_t run_filtering(std::string sInputFilePath, std::string sOutputFilePath, CCombFilterIf::CombFilterType_t filterType, float gain, float delayInSeconds, int blockSize)
```

Comment:
A little awkward to have this in a header file; would prefer `.cpp`. (Either included in `MUSI6106Exec.cpp` directly or in its own `.cpp` file.) Ditto for `tests.h`; generally preferable to mostly stick to declarations in header files.

Reply:
I don't fully understand, could you clarify what you mean here please? @ijc8

Reply:
Sure. I just meant that header files are usually used for _declarations_ (e.g. function signatures, class declarations, enums, etc.), while definitions (the actual implementation) go in ordinary source files (`.cpp`). Thus, I was surprised to see these non-trivial function definitions in header files (`run_filtering.h` and `tests.h`) as opposed to source files (`run_filtering.cpp` and `tests.cpp`).

(There are exceptions, like when an implementation is particularly short, as with most getters and setters, or if you're doing a lot with templates, but I don't think those apply here.)